frankenphp.go

```diff
@@ -486,20 +495,26 @@ func go_write_header(rh C.uintptr_t, status C.int) {
 }
 
 //export go_sapi_flush
-func go_sapi_flush(rh C.uintptr_t) {
+func go_sapi_flush(rh C.uintptr_t) bool {
 	r := cgo.Handle(rh).Value().(*http.Request)
 	fc := r.Context().Value(contextKey).(*FrankenPHPContext)
 
 	if fc.responseWriter == nil {
-		return
+		return true
 	}
 
 	flusher, ok := fc.responseWriter.(http.Flusher)
 	if !ok {
-		return
+		return true
+	}
+
+	if clientHasClosed(r) {
+		return true
 	}
 
 	flusher.Flush()
```

Comment:
Since `Flush()` on a closed connection seems to be a noop, I'm inclined to remove the previous if-statement and just return the result of `clientHasClosed()`. What are your thoughts on that?

Reply:
It's definitely a micro-optim, but doing this in this order prevents a useless call to `Flush()` event if it's a noop. I don't have a strong opinion on this TBH.

Reply:
True. I don't have strong feelings either, so I'll leave as-is. My optimization thought was mostly to reduce the branching, but it's probably better to leave that sort of optimization for a compiler and just describe what is correct.